dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java

```diff
@@ -19,16 +19,44 @@
 
 import uk.ac.ebi.eva.commons.core.models.IVariant;
 import uk.ac.ebi.eva.commons.core.models.VariantCoreFields;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
 import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
 
+/**
+ * TODO Create copy of this class for variants submitted by EVA, which don't need a nested VariantSourceEntry object
+ */
 public class SubSnpCoreFieldsToVariantProcessor implements ItemProcessor<SubSnpCoreFields, IVariant> {
 
+    public static final String DBSNP_BUILD_KEY = "dbSNP build";
+
+    private final String dbsnpBuild;
+
+    private final String batch;
+
+    public SubSnpCoreFieldsToVariantProcessor(int dbsnpBuild, int batch) {
+        this.dbsnpBuild = String.valueOf(dbsnpBuild);
+        this.batch = String.valueOf(batch);
+    }
+
     @Override
     public IVariant process(SubSnpCoreFields subSnpCoreFields) throws Exception {
         VariantCoreFields variantCoreFields = subSnpCoreFields.getVariantCoreFields();
-        
-        return null;
-    }
 
+        Variant variant = new Variant(variantCoreFields.getChromosome(), variantCoreFields.getStart(),
+                                      variantCoreFields.getEnd(), variantCoreFields.getReference(),
+                                      variantCoreFields.getAlternate());
+        // Set current 'rs' as main variant ID
+        if (subSnpCoreFields.getRsId() != null) {
+            variant.setMainId("rs" + subSnpCoreFields.getRsId());
+        }
+
+        VariantSourceEntry variantSourceEntry = new VariantSourceEntry(batch, batch);
+        variantSourceEntry.addAttribute(DBSNP_BUILD_KEY, dbsnpBuild);
+        variantSourceEntry.setSecondaryAlternates(subSnpCoreFields.getSecondaryAlternatesInForwardStrand());
```

Comment:
I would agree to extract `getSecondaryAlternatesInForwardStrand()` from the bean, and putting it here as well. It just seemed to belong next to getAllelesInForwardStrand, or getVariantCoreFields.

Reply:
The way it is right now is fine for me.